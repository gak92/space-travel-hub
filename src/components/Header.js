import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from '../images/planet.png';

const Header = () => (
  <div className="header">
    <div className="logo-div">
      <img src={planet} alt="space" className="logo-image" />
      <h1>Space Travelers hub</h1>
    </div>
    <nav className="navbar">
      <NavLink to="/">Rockets</NavLink>
      <NavLink to="/missions">Missions</NavLink>
      <NavLink to="/myprofile">MyProfile</NavLink>
    </nav>
  </div>
);

export default Header;
