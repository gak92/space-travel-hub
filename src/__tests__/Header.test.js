import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Header from "../components/Header";

it('Header renders correctly', () => {
  const tree = render(
    <Router>
      <Header />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});

it('Header user interaction', () => {
  render(
    <Router>
      <Header />
    </Router>
  );

  const Element = screen.getByText('Space Travelers hub');
  expect(Element).toBeInTheDocument();
});