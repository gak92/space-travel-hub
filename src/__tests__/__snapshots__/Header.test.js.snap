// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="logo-div"
        >
          <img
            alt="space"
            class="logo-image"
            src="planet.png"
          />
          <h1>
            Space Travelers hub
          </h1>
        </div>
        <nav
          class="navbar"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
          <a
            class=""
            href="/myprofile"
          >
            MyProfile
          </a>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="logo-div"
      >
        <img
          alt="space"
          class="logo-image"
          src="planet.png"
        />
        <h1>
          Space Travelers hub
        </h1>
      </div>
      <nav
        class="navbar"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Rockets
        </a>
        <a
          class=""
          href="/missions"
        >
          Missions
        </a>
        <a
          class=""
          href="/myprofile"
        >
          MyProfile
        </a>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
